src/python/evaluation/plots/diffs_plotter.py

```diff
@@ -0,0 +1,167 @@
+import argparse
+import sys
+from pathlib import Path
+from statistics import median
+from typing import Any, Callable, Dict
+
+sys.path.append('../../../..')
+
+import pandas as pd
+import plotly.graph_objects as go
+from src.python.common.tool_arguments import RunToolArgument
+from src.python.evaluation.inspectors.common.statistics import IssuesStatistics, PenaltyInfluenceStatistics
+from src.python.evaluation.inspectors.print_inspectors_statistics import gather_statistics
+from src.python.evaluation.plots.common import create_bar_plot, create_box_plot, Extension, save_plot
+from src.python.review.common.file_system import deserialize_data_from_file
+
+
+def configure_arguments(parser: argparse.ArgumentParser) -> None:
+    parser.add_argument(
+        RunToolArgument.DIFFS_FILE_PATH.value.long_name,
+        type=lambda value: Path(value).absolute(),
+        help=RunToolArgument.DIFFS_FILE_PATH.value.description,
+    )
+
+    parser.add_argument(
+        "save_dir",
+        type=lambda value: Path(value).absolute(),
+        help="The directory where the plotted charts will be saved",
+    )
+
+    parser.add_argument(
+        "--file-extension",
+        type=str,
+        default=Extension.SVG.value,
+        choices=Extension.values(),
+        help="Allows you to select the extension of output files",
+    )
+
+
+def _get_dataframe_from_dict(
+    data_dict: Dict[Any, Any],
+    key_name: str,
+    value_name: str,
+    key_mapper: Callable = lambda x: x,
+    value_mapper: Callable = lambda y: y,
+):
+    """
+    Converts 'data_dict' to a dataframe consisting of two columns: 'key_name', 'value_name'.
+    'key_name' contains all keys of 'data_dict', 'value_name' contains all corresponding
+    values of 'data_dict'. With the functions 'key_mapper' and 'value_mapper' you can
+    additionally convert keys and values respectively.
+    """
+    converted_dict = {
+        key_name: list(map(key_mapper, data_dict.keys())),
+        value_name: list(map(value_mapper, data_dict.values())),
+    }
+
+    return pd.DataFrame.from_dict(converted_dict)
+
+
+def get_unique_issues_by_category(
+    statistics: IssuesStatistics,
+    x_axis_name: str = "Categories",
+    y_axis_name: str = "Number of unique issues",
+    limit: int = 0,
+) -> go.Figure:
+    categorized_statistics = statistics.get_short_categorized_statistics()
+    filtered_stats = {issue_type: stat[0] for issue_type, stat in categorized_statistics.items() if stat[0] >= limit}
+
+    df = _get_dataframe_from_dict(
+        filtered_stats,
+        key_name=x_axis_name,
+        value_name=y_axis_name,
+        key_mapper=lambda issue_type: issue_type.name,
+    )
+
+    return create_bar_plot(df, x_axis_name, y_axis_name)
+
+
+def get_issues_by_category(
```

Comment:
This function is very simular into get_unique_issues_by_category. Can we create a new one with a common part and reuse it?